// Copyright 2023 The Tint Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package tint.core.ir.binary.pb;

message Module {
    repeated TypeDecl types = 1;
    Block root_block = 2;
    repeated Function functions = 3;
}

message Block {
}

message Function {
    optional string name = 1;
    uint32 return_type = 2;
    optional PipelineStage pipeline_stage = 3;
}

enum PipelineStage {
    Compute = 0;
    Fragment = 1;
    Vertex = 2;
}

message TypeDecl {
    oneof kind {
        ScalarType scalar = 1;
    }
}

enum ScalarType {
    bool = 0;
    i32 = 1;
    u32 = 2;
    f32 = 3;
}
